src/checkers/inference/InferenceAnnotatedTypeFactory.java

```diff
@@ -389,32 +391,15 @@ public ParameterizedExecutableType constructorFromUse(final NewClassTree newClas
                                       "Current path:\n" + getVisitorTreePath();
 
         final ExecutableElement constructorElem = TreeUtils.elementFromUse(newClassTree);
-        @SuppressWarnings("deprecation") // TODO
-        final AnnotatedTypeMirror constructorReturnType = fromNewClass(newClassTree);
-        addComputedTypeAnnotations(newClassTree, constructorReturnType);
 
-        final AnnotatedExecutableType constructorType = AnnotatedTypes.asMemberOf(types, this, constructorReturnType, constructorElem);
-        // Take adapt parameter logic from AnnotatedTypeFactory#constructorFromUse to
-        // InferenceAnnotatedTypeFactory#constructorFromUse.
-        // Store varargType before calling setParameterTypes, otherwise we may lose the
-        // varargType as it is the last element of the original parameterTypes.
-        // AnnotatedTypes.asMemberOf handles vararg type properly, so we do not need to compute
-        // vararg type again.
-        constructorType.computeVarargType();
-        // Adapt parameters, which makes parameters and arguments be the same size for later
-        // checking. The vararg type of con has been already computed and stored when calling
-        // typeVarSubstitutor.substitute.
-        List<AnnotatedTypeMirror> parameters =
-                AnnotatedTypes.adaptParameters(this, constructorType, newClassTree.getArguments());
-        constructorType.setParameterTypes(parameters);
-        if (viewpointAdapter != null) {
-            viewpointAdapter.viewpointAdaptConstructor(constructorReturnType, constructorElem, constructorType);
-        }
+        AnnotatedDeclaredType constructorReturnType =
```

Comment:
Can you add some comments, e.g. why are we computing and annotating `constructorReturnType`, even though it's not used later in the method.

Reply:
Updated!

Reply:
> Why is this only necessary for the return type, not the parameter types? The old code also did something with the parameters.

Because other code is in the super call now.

Reply:
Which leaves the question why the supermethod isn't doing that... but okay, let's just do it only here.

Reply:
I leave a todo there for future clean up.